roles/vm-spinup/tasks/main.yml

```diff
@@ -12,32 +12,26 @@
     mode: 0755
 
 - name: Get the virsh list
-  shell: >
+  command: >
     virsh list
   register: virsh_list
+  when: virsh_list not defined
 
 # - name: Show virsh list
 #   debug: "msg={{ virsh_list.stdout }} - {{ virtual_machines }}"
 
 - name: Run spinup for each host that doesn't exist
-  shell: >
+  command: >
     /root/spinup.sh {{ item.name }} {{ item.system_ram_mb | default(system_default_ram_mb) }} {{ item.system_cpus | default(system_default_cpus) }}
   when: "item.name not in virsh_list.stdout"
   with_items: "{{ virtual_machines }}"
 
-# - name: Get IP Addresses for all VMs
-#   shell: >
-#     grep -B1 $(virsh dumpxml {{ item }} | awk -F\' '/mac address/ {print $2}') /var/lib/libvirt/dnsmasq/{{ bridge_name }}.status | head -n 1 | awk '{print $2}' | sed -e s/\"//g -e s/,//
-#   register: vm_ip_addresses
-#   with_items: "{{ virtual_machines }}"
-#   when: not bridge_networking
-
 - name: Get IP Addresses for all VMs
-  shell: >
+  command: >
     cat /tmp/{{ item.name }}.ip.txt
   register: vm_ip_addresses
   with_items: "{{ virtual_machines }}"
-  # when: bridge_networking
+  when: vm_ip_addresses not defined
```

Comment:
JW, where else is `vm_ip_addresses` -- or maybe in case this gets called multiple times?

Reply:
It's not called anywhere else actually. This is somewhat of a "workaround" to ansible-lint complaining when you use `command` or `shell` without a `when` or `when_changed` etc.

Reply:
The alternative is to actually do:

```
tags:
  - skip_ansible_lint
  # Skip ANSIBLE0012 because we need to run this every time
```